src/pages/Dashboard/Modals/SupplyWithdraw/index.tsx

```diff
@@ -92,69 +123,81 @@ export const SupplyWithdrawUi: React.FC<ISupplyWithdrawUiProps> = ({
-                      calculateNewBalance={calculateNewBorrowAmount}
-                    />,
-                  ][activeTabIndex]
-                }
-              </EnableToken>
-            )}
+            <EnableToken
+              symbol={assetId as TokenId}
+              title={activeTabTextKeys.enable}
+              tokenInfo={tokenInfo}
+              isEnabled={!!isEnabled}
+              vtokenAddress={asset.vtokenAddress}
+            >
+              {
+                [
+                  <SupplyWithdrawForm
+                    key="supply"
+                    asset={asset}
+                    tokenInfo={tokenInfo}
+                    userTotalBorrowBalance={userTotalBorrowBalance}
+                    userTotalBorrowLimit={userTotalBorrowLimit}
+                    dailyEarnings={dailyEarnings}
+                    onSubmit={onSubmitSupply}
+                    inputLabel={t('supplyWithdraw.walletBalance')}
+                    enabledButtonKey={t('supplyWithdraw.supply')}
+                    disabledButtonKey={t('supplyWithdraw.enterValidAmountSupply')}
+                    maxInput={asset.walletBalance}
+                    calculateNewBalance={calculateNewSupplyAmount}
+                    isTransactionLoading={isTransactionLoading}
+                  />,
+                  <SupplyWithdrawForm
+                    key="withdraw"
+                    asset={asset}
+                    tokenInfo={tokenInfo}
+                    userTotalBorrowBalance={userTotalBorrowBalance}
+                    userTotalBorrowLimit={userTotalBorrowLimit}
+                    dailyEarnings={dailyEarnings}
+                    onSubmit={onSubmitWithdraw}
+                    inputLabel={t('supplyWithdraw.withdrawableAmount')}
+                    enabledButtonKey={t('supplyWithdraw.withdraw')}
+                    disabledButtonKey={t('supplyWithdraw.enterValidAmountWithdraw')}
+                    maxInput={asset.supplyBalance}
+                    calculateNewBalance={calculateNewBorrowAmount}
+                    isTransactionLoading={isTransactionLoading}
+                  />,
+                ][activeTabIndex]
+              }
+            </EnableToken>
           </ConnectWallet>
         </div>
       </>
     </Modal>
   );
 };
 
-const SupplyWithdrawModal: React.FC<
-  Omit<ISupplyWithdrawUiProps, 'userTotalBorrowBalance' | 'userTotalBorrowLimit' | 'dailyEarnings'>
-> = props => {
+const SupplyWithdrawModal: React.FC<ISupplyWithdrawUiProps> = props => {
+  const { asset, ...rest } = props;
   const { account } = useContext(AuthContext);
   const { userTotalBorrowBalance, userTotalBorrowLimit } = useUserMarketInfo({
     account: account?.address,
   });
+  const { supply, isLoading: isSupplyLoading } = useSupply({ asset, account: account?.address });
+  const {
+    redeem,
+    redeemUnderlying,
+    vTokenBalance,
+    isLoading: isWithdrawLoading,
+  } = useWithdraw({ asset, account: account?.address });
```

Comment:
We need to handle errors as well. It doesn't have to be complicated, we can just display them in a toast, but we need to give an indication to users when a transaction fails.